jwala-services/src/test/java/com/cerner/jwala/service/jvm/impl/JvmServiceImplTest.java

```diff
@@ -1281,5 +1281,56 @@ public JvmService getJvmService() {
+        when(mockJvm.getId()).thenReturn(new Identifier<Jvm>(111L));
+        when(mockJvm.getJdkMedia()).thenReturn(new Media(1L, "test JDK media", MediaType.JDK,
+                Paths.get("x:/test/archive/path.zip"), Paths.get("x:/test-destination"), Paths.get("root-dir-destination")));
+        when(mockJvm.getTomcatMedia()).thenReturn(new Media(2L, "test Tomcat media", MediaType.TOMCAT,
+                Paths.get("./src/test/resources/binaries/apache-tomcat-test.zip"), Paths.get("x:/test-destination-tomcat"), Paths.get("tomcat-root-dir-destination")));
+        when(commandOutput.getReturnCode()).thenReturn(new ExecReturnCode(0));
+        when(Config.mockJvmControlService.secureCopyFile(any(ControlJvmRequest.class), anyString(), anyString(), anyString(), anyBoolean())).thenReturn(commandOutput);
+        when(Config.mockJvmControlService.executeCreateDirectoryCommand(any(Jvm.class), anyString())).thenReturn(commandOutput);
+        when(Config.mockJvmControlService.executeChangeFileModeCommand(any(Jvm.class), anyString(), anyString(), anyString())).thenReturn(commandOutput);
+        when(Config.mockJvmControlService.controlJvm(eq(ControlJvmRequestFactory.create(JvmControlOperation.DELETE_SERVICE, mockJvm)), any(User.class))).thenReturn(commandOutput);
+        when(Config.mockJvmControlService.controlJvm(eq(ControlJvmRequestFactory.create(JvmControlOperation.DEPLOY_JVM_ARCHIVE, mockJvm)), any(User.class))).thenReturn(commandOutput);
+        when(Config.mockJvmControlService.controlJvm(eq(ControlJvmRequestFactory.create(JvmControlOperation.INSTALL_SERVICE, mockJvm)), any(User.class))).thenReturn(commandOutput);
+        when(Config.mockJvmControlService.executeCheckFileExistsCommand(any(Jvm.class), anyString())).thenReturn(commandOutput);
+
+        when(Config.mockJvmPersistenceService.findJvmByExactName(anyString())).thenReturn(mockJvm);
+        when(Config.mockJvmPersistenceService.getJvmTemplate(anyString(), any(Identifier.class))).thenReturn("<server>some xml</server>");
+
+        when(Config.mockResourceService.generateResourceGroup()).thenReturn(mockResourceGroup);
+        when(Config.mockResourceService.generateResourceFile(anyString(), anyString(), any(ResourceGroup.class), anyObject(), any(ResourceGeneratorType.class))).thenReturn("<server>some xml</server>");
+
+        final List<String> templateNames = new ArrayList<>();
+        templateNames.add("setenv.bat");
+        when(Config.mockJvmPersistenceService.getResourceTemplateNames(anyString())).thenReturn(templateNames);
+
+        Jvm response = jvmService.upgradeJDKAndDeployJvm(mockJvm.getJvmName(), Config.mockUser);
+        assertEquals(response.getJvmName(), mockJvm.getJvmName());
+
+        when(mockJvm.getState()).thenReturn(JvmState.JVM_NEW);
+        boolean exceptionThrown = false;
+        try {
+            jvmService.upgradeJDKAndDeployJvm(mockJvm.getJvmName(), Config.mockUser);
+        } catch (Exception e) {
+            exceptionThrown = true;
```

Comment:
When you make this a separate test you can use the `@Test(expected = Exception.class` to test for exceptions being thrown instead of setting and checking a flag.